test/proposal.test.js

```diff
@@ -0,0 +1,194 @@
+  info: {
+    about: {
+      shortDescription: 'this is short description',
+      longDescription: 'this is a very long description',
+      website: 'https://www.google.com',
+      designation: 'software engg',
+      skills: ['c++', 'java'],
+      education: [
+        {
+          school: {
+            schoolName: 'firstSchoolName',
+            year: '2017-2021'
+          }
+        },
+        {
+          school: {
+            schoolName: 'secondSchoolName',
+            year: '2007-2014'
+          }
+        }
+      ],
+      location: 'location'
+    }
+  }
+}
+
+const demoOrganization = {
+  name: 'Codeuino',
+  description: {
+    shortDescription: 'short desc',
+    longDescription: 'long Description included here'
+  },
+  contactInfo: {
+    email: 'organisation@test.com',
+    website: 'www.codeuino.org'
+  }
+}
+
+const testOrganization = {
+  _id: testOrganizationId,
+  ...demoOrganization
+}
+
+const updatedProposalContent = {
+  content: 'updated proposal content'
+}
+
+const testUser = {
+  _id: testUserId,
+  ...demoUser,
+  email: `test${randomDigit}@mailinator.com`,
+  phone: `12345678${randomDigit}`,
+  tokens: [
+    {
+      token: jwt.sign(
+        {
+          _id: testUserId
+        },
+        process.env.JWT_SECRET
+      )
+    }
+  ]
+}
+
+let server
+
+/**
+ * This will pe performed once at the beginning of the test
+ */
+
+beforeAll(async (done) => {
+  await Proposal.deleteMany()
+  await new User(testUser).save()
+  await new Organization(testOrganization).save()
+  server = app.listen(4000, () => {
+    global.agent = request.agent(server)
+  })
+  const response = await request(app).post('/auth/login').send({
+    email: testUser.email,
+    password: testUser.password
+  })
+  token = response.body.token
+  done()
+})
+
+/**
+ * This deletes all the existing user in database,
+ * and creates a new user in database with the provided details.
+ */
+beforeEach(async () => {
+  await Proposal.deleteMany()
+  await new Proposal(testProposal).save()
+})
+
+test('Should create new Proposal', async (done) => {
+  const response = await request(app)
+    .post('/proposal')
+    .set('Authorization', `Bearer ${token}`)
+    .send(demoproposal)
+    .expect(HttpStatus.CREATED)
+
+  const proposal = await Proposal.findById(response.body.proposal._id)
+  expect(proposal).not.toBeNull()
+
+  const userId = response.body.proposal.creator
+
+  expect(response.body).toMatchObject({
+    proposal: {
+      title: demoproposal.title,
+      organization: `${testOrganizationId}`,
+      content: demoproposal.content,
+      proposalStatus: demoproposal.proposalStatus,
+      creator: `${userId}`
+    }
+  })
+  done()
+})
+
+// Testing proposal update
+test('Should update the content of the proposal', async (done) => {
+  await request(app)
+    .patch(`/proposal/${testProposalId}`)
+    .set('Authorization', `Bearer ${token}`)
+    .send(updatedProposalContent)
+    .expect(HttpStatus.OK)
+
```

Comment:
check if the proposal collections is updated or not?

Reply:
@Rupeshiya Will be fixing this and the other issue on the bottom by Monday. I'm working on writing the remaining test cases as well.